media/server/gstplayer/source/GstGenericPlayer.cpp

```diff
@@ -339,8 +373,9 @@ bool GstGenericPlayer::getPosition(std::int64_t &position)
 {
     // We are on main thread here, but m_context.pipeline can be used, because it's modified only in GstGenericPlayer
     // constructor and destructor. GstGenericPlayer is created/destructed on main thread, so we won't have a crash here.
-    if (!m_context.pipeline || GST_STATE(m_context.pipeline) < GST_STATE_PLAYING)
+    if (!m_context.pipeline || GST_STATE(m_context.pipeline) < GST_STATE_PAUSED)
```

Comment:
Is this an intentional change, it doesn't seem related to stream switching

Reply:
Yes, it is. performAudioTrackCodecChannelSwitch has to check the playback position in PAUSED state. According to GStreamer documentation, it allows to check current position in > READY state (so in PAUSED and PLAYING). I checked all YT tests, this change does not break anything.